scripts/pre_commit_linter.py

```diff
@@ -1280,39 +1280,16 @@ def _check_js_component_name_and_count(self):
                     component_name = arguments[0].value
                     component = expression.callee.property.name
 
-                    # If the component is directive or filter and its name is
```

Comment:
Why delete all this? We should update the check so that it works for the new convention.

Reply:
@seanlip As we have discussed earlier, this sample PR was to go in first. So if I employed the check for the new convention then the rest of the files would fail it, and the only solution would be to check the filenames for two conventions which seemed a little weird to me. Therefore I removed the filename check only to be updated when the refactoring is complete.

Reply:
Ah. This is an issue. I'm fine with whatever PR strategy you choose, but these checks cannot be removed, even in the interim. Either swap it out entirely (as a result of a large PR) or maintain a whitelist of "not-yet-migrated" files that undergo the old check (while the migrated files undergo the new check). But we should never reduce the coverage of the checks.

Reply:
Actually, I was thinking of adding the correct check in that one large PR. This way the other files will not fail the CI builds here and will also reduce the need of adding two convention checks.

Reply:
That's fine, as long as we never merge a PR into develop that reduces the checks.